### PR TITLE
TINY-8241: Removed all Promise polyfills

### PR DESCRIPTION
**Pull request by HAFRMO**
Related Ticket:

Description of Changes:
Removed Polyfill promises ( We don't seem to have any other polyfill things in the tinymce repo, though we do rely on things with polyfill stuff ).

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* N/A Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/agar/.eslintrc.json [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "rules": {
+    "@tinymce/no-unimported-promise": "off",
```
Comment (lnewson) [inline modules/agar/.eslintrc.json:3]: This isn't needed here as it's in the root file instead. These files only exist as overrides to the root file.

File: modules/alloy/.eslintrc.json [hunk 2]
```diff
@@ -1,5 +1,6 @@
 {
   "rules": {
+    "@tinymce/no-unimported-promise": "off",
```
Comment (lnewson) [inline modules/alloy/.eslintrc.json:3]:
```suggestion
```
Likewise this isn't needed here.

File: modules/alloy/CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 - Upgraded to Katamari 9.0, which includes breaking changes to the `Optional` API used in this module.
+- Removed usage of polyfill promises in favor of native promises.
```
Comment (lnewson) [inline modules/alloy/CHANGELOG.md:11]: There are no changes in alloy so should this be here?

File: modules/agar/CHANGELOG.md [hunk 4]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 - Upgraded to Katamari 9.0, which includes breaking changes to the `Optional` API used in this module.
+- Removed usage of polyfill promises in favor of native promises.
```
Comment (lnewson) [inline modules/agar/CHANGELOG.md:11]: This is a removed entry so it should not be under the `Changed` heading.

File: modules/imagetools/CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 - Upgraded to Katamari 9.0, which includes breaking changes to the `Optional` API used in this module.
+- Removed usage of polyfill promises in favor of native promises.
```
Comment (lnewson) [inline modules/imagetools/CHANGELOG.md:11]: Likewise this should be under the `Removed` heading.

File: modules/katamari-assertions/CHANGELOG.md [hunk 6]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 - Updated to Katamari 9.0, which is incompatible with Katamari 8.0 if used in the same bundle.
+- Removed usage of polyfill promises in favor of native promises.
```
Comment (lnewson) [inline modules/katamari-assertions/CHANGELOG.md:11]: Likewise this should be under the `Removed` heading.

File: modules/tinymce/.eslintrc.json [hunk 7]
```diff
@@ -1,6 +1,7 @@
 {
   "plugins":[ "notice" ],
   "rules": {
+    "@tinymce/no-unimported-promise": "off",
```
Comment (lnewson) [inline modules/tinymce/.eslintrc.json:4]:
```suggestion
```
Likewise this isn't needed here.

File: modules/katamari/CHANGELOG.md [hunk 8]
```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Changed
 - Swapped `Optional` to be a class, instead of an object with function fields.
+- Removed usage of polyfill promises in favor of native promises.
```
Comment (lnewson) [inline modules/katamari/CHANGELOG.md:14]: Likewise this should be under the `Removed` heading.

File: modules/tinymce/CHANGELOG.md [hunk 9]
```diff
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The `editor.setContent()` API can now be prevented using the `BeforeSetContent` event. This makes it consistent with the `editor.selection.setContent()` API #TINY-8018
 - Aligning a table to the left or right will now use margin styling instead of float styling #TINY-6558
 - The `tinymce.settings` global property is no longer set upon initialization #TINY-7359
+- Removed usage of polyfill promises in favor of native promises #TINY-8241
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:17]: This falls under the `Removed support for Microsoft Internet Explorer 11` changelog entry that already exists so please just add the Jira number there instead.

File: modules/tinymce/CHANGELOG.md [hunk 10]
```diff
@@ -38,6 +39,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Removed the deprecated `fullpage`, `spellchecker`, `bbcode`, `legacyoutput`, `colorpicker`, `contextmenu` and `textcolor` plugins #TINY-8192
 - Removed support for Word from the `paste` plugin #TINY-7493
 - Removed the `imagetools` plugin, which is now classified as a Premium plugin #TINY-8209
+- Removed the `tinymce.utils.Promise` from API #TINY-8241
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:42]:
```suggestion
- Removed the `tinymce.utils.Promise` API #TINY-8241
```

File: modules/tinymce/src/core/main/ts/api/Tinymce.ts [hunk 11]
```diff
@@ -102,7 +101,6 @@ interface TinyMCE extends EditorManager {
   };
 
   util: {
-    Promise: PromiseConstructor;
```
Comment (lnewson) [inline modules/tinymce/src/core/main/ts/api/Tinymce.ts:105]: You've missed the `util.Promise` variable here on line 207.

**Comment by lnewson** [comment]
I had a quick look through and couldn't see anything else that had been missed thanks @HAFRMO!

**Comment by TheSpyder** [comment]
Looks good, once Lee's comments are addressed we can move to PR.

**Review by ashfordneil — APPROVED** [review]

**Review by lnewson — APPROVED** [review]

**Pull request merged**